tmt/utils/__init__.py

```diff
@@ -2446,28 +2461,105 @@ def from_env(cls) -> 'TracebackVerbosity':
 
 
 def render_run_exception_streams(
-        stdout: Optional[str],
-        stderr: Optional[str],
-        verbose: int = 0) -> Iterator[str]:
+        output: CommandOutput,
+        verbose: int = 0,
+        comment_sign: str = '#') -> Iterator[str]:
     """ Render run exception output streams for printing """
 
-    for name, output in (('stdout', stdout), ('stderr', stderr)):
-        if not output:
+    for name, content in (('stdout', output.stdout), ('stderr', output.stderr)):
+        if not content:
             continue
-        output_lines = output.strip().split('\n')
+        content_lines = content.strip().split('\n')
         # Show all lines in verbose mode, limit to maximum otherwise
         if verbose > 0:
-            line_summary = f"{len(output_lines)}"
+            line_summary = f"{len(content_lines)}"
         else:
-            line_summary = f"{min(len(output_lines), OUTPUT_LINES)}/{len(output_lines)}"
-            output_lines = output_lines[-OUTPUT_LINES:]
+            line_summary = f"{min(len(content_lines), OUTPUT_LINES)}/{len(content_lines)}"
+            content_lines = content_lines[-OUTPUT_LINES:]
+
+        line_intro = f'{comment_sign} '
+
+        yield line_intro + f'{name} ({line_summary} lines)'
+        yield line_intro + (OUTPUT_WIDTH - 2) * '~'
+        yield from content_lines
+        yield line_intro + (OUTPUT_WIDTH - 2) * '~'
+        yield ''
+
+
+@overload
+def render_command_report(
+        *,
+        label: str,
+        command: Optional[Union[ShellScript, Command]] = None,
+        output: CommandOutput,
+        exc: None = None) -> Iterator[str]:
+    pass
+
+
+@overload
+def render_command_report(
+        *,
+        label: str,
+        command: Optional[Union[ShellScript, Command]] = None,
+        output: None = None,
+        exc: RunError) -> Iterator[str]:
+    pass
+
+
+def render_command_report(
+        *,
+        label: str,
+        command: Optional[Union[ShellScript, Command]] = None,
+        output: Optional[CommandOutput] = None,
+        exc: Optional[RunError] = None,
+        comment_sign: str = '#') -> Iterator[str]:
+    """
+    Format a command output for a report file.
+
+    To provide unified look of various files reporting command outputs,
+    this helper would combine its arguments and emit lines the caller
+    may then write to a file. The following template is used:
+
+    .. code-block::
 
-        yield f'{name} ({line_summary} lines)'
-        yield OUTPUT_WIDTH * '~'
-        yield from output_lines
-        yield OUTPUT_WIDTH * '~'
+        ## ${label}
+
+        # ${command}
+
+        # stdout (N lines)
+        #~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
```suggestion
        # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Here's a couple more in the docstring.

Reply:
Addressed in 7fa0e284